azerothcore/azerothcore-wotlk
# fix (core): EffectInstaKill Prevents spell animation fix

This fixes the issue of spell animations of chain effects of a spell dealing with EffectInstaKill. This was identified as a issue in support of @malcrom 's current scripting efforts. We fix the issue by putting a half second delay and making the delay of a function possible and independent.

Co-Authored-By: Malcrom <malcromdev@gmail.com>



## Changes Proposed:
-  Add independent Delay Feature in SpellEffect.cpp

## Issues Addressed:

- Closes 

## SOURCE:

Sniff Team

## Tests Performed:

- Builds
- Performs

## How to Test the Changes:


1. Currently eith the pr do a .cast back 33974 on a creature and you will see the power burn happen follow by a bomb explosion (shockwave effect) then instakill.
2. without this pr you will not see the bomb explosion shockwave effect

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
